paperqa/contrib/zotero.py

```diff
@@ -107,6 +127,22 @@ def get_pdf(self, item: dict) -> Union[Path, None]:
         pdf_path = self.storage / (pdf_key + ".pdf")
 
         if not pdf_path.exists():
+            if self.zotero_storage:
+                self.logger.info(f"|  Looking for existing PDF for: {self._get_citation_key(item)}")
+                try:
+                    zotero_doc_folder = self.zotero_storage / pdf_key
+
+                    if zotero_doc_folder.exists():
+                        pdf_files = list(zotero_doc_folder.glob("*.pdf"))
+                        if len(pdf_files) == 1:
+                            self.logger.info(f"|  Copying existing PDF for {self._get_citation_key(item)} from Zotero storage.")
+                            zotero_pdf_path = zotero_doc_folder / pdf_files[0]
+                            shutil.copy(zotero_pdf_path, pdf_path)
+                            return pdf_path
+
+                except Exception as e:
```

Comment:
What other exceptions are you thinking of here? Maybe catch them explicitly, and throw the error if something unexpected comes up?

Reply:
Are you suggesting one of the three things below? Or something else?

catch specific error cases and resolve them
catch, warn and fall back to downloading
catch, wrap with a message, and raise

For unexpected edge cases, do you think raising an error would be better or just warning the user and falling back to downloading?

Reply:
Didn't have anything specific in mind - could imagine permissions errors at the destination dir.

Reply:
If there was no specific error you had in mind I would just remove the try-except. If you think there could be permissions issues for the `shutil.copy` I would check for the specific permissions error, and warn, then fall back to downloading. If there's an unexpected error then it should be raised.